src/Microsoft.DotNet.XHarness.TestRunners.Xunit/ThreadlessXunitTestRunner.cs

```diff
@@ -0,0 +1,137 @@
+            discoverySink.Finished.WaitOne();
+            var testCasesToRun = discoverySink.TestCases.Where(filters.Filter).ToList();
+            Console.WriteLine($"Discovery finished.");
+
+            var summarySink = new DelegatingExecutionSummarySink(testSink, () => false, (completed, summary) => { Console.WriteLine($"Tests run: {summary.Total}, Errors: 0, Failures: {summary.Failed}, Skipped: {summary.Skipped}{Environment.NewLine}Time: {TimeSpan.FromSeconds((double)summary.Time).TotalSeconds}s"); });
+            var resultsXmlAssembly = new XElement("assembly");
+            var resultsSink = new DelegatingXmlCreationSink(summarySink, resultsXmlAssembly);
+
+            testSink.Execution.TestPassedEvent += args => { Console.WriteLine($"[PASS] {args.Message.Test.DisplayName}"); };
+            testSink.Execution.TestSkippedEvent += args => { Console.WriteLine($"[SKIP] {args.Message.Test.DisplayName}"); };
+            testSink.Execution.TestFailedEvent += args => { Console.WriteLine($"[FAIL] {args.Message.Test.DisplayName}{Environment.NewLine}{ExceptionUtility.CombineMessages(args.Message)}{Environment.NewLine}{ExceptionUtility.CombineStackTraces(args.Message)}"); };
+
+            testSink.Execution.TestAssemblyStartingEvent += args => { Console.WriteLine($"Running tests for {args.Message.TestAssembly.Assembly}"); };
+            testSink.Execution.TestAssemblyFinishedEvent += args => { Console.WriteLine($"Finished {args.Message.TestAssembly.Assembly}{Environment.NewLine}"); };
+
+            controller.RunTests(testCasesToRun, resultsSink, testOptions);
+            resultsSink.Finished.WaitOne();
+
+            if (printXml)
+            {
+                Console.WriteLine($"STARTRESULTXML");
+                var resultsXml = new XElement("assemblies");
+                resultsXml.Add(resultsXmlAssembly);
+                Console.WriteLine(resultsXml.ToString());
+                Console.WriteLine();
+                Console.WriteLine($"ENDRESULTXML");
+            }
+
+            var failed = resultsSink.ExecutionSummary.Failed > 0 || resultsSink.ExecutionSummary.Errors > 0;
+            return failed ? 1 : 0;
+        }
+
+        void ParseEqualSeparatedArgument(Dictionary<string, List<string>> targetDictionary, string argument)
+        {
+            var parts = argument.Split('=');
+            if (parts.Length != 2 || string.IsNullOrEmpty(parts[0]) || string.IsNullOrEmpty(parts[1]))
+            {
+                throw new ArgumentException("Invalid argument value '{argument}'.", nameof(argument));
+            }
+
+            var name = parts[0];
+            var value = parts[1];
+            List<string> excludedTraits;
+            if (targetDictionary.TryGetValue(name, out excludedTraits!))
+            {
+                excludedTraits.Add(value);
+            }
+            else
+            {
+                targetDictionary[name] = new List<string> { value };
+            }
```

Comment:
We already have this logic in the Common namespace, why re-doing it? Should we move it to a re-usable file?